web/src/models/DataStore.model.ts

```diff
@@ -1,41 +1,35 @@
 import {SupportedDataStores} from 'types/DataStore.types';
 import {Model, TDataStoreSchemas} from 'types/Common.types';
 
-export type TRawDataStore = TDataStoreSchemas['DataStore'];
-type DataStore = Model<
-  TRawDataStore,
-  {
-    otlp?: {};
-    newRelic?: {};
-    lightstep?: {};
-    datadog?: {};
-  }
->;
+export type TRawDataStore = TDataStoreSchemas['DataStoreResource'];
+type DataStore = Model<TRawDataStore, {}>['spec'] & {otlp?: {}; newRelic?: {}; lightstep?: {}; datadog?: {}};
 
 const DataStore = ({
-  id = '',
-  name = '',
-  type = SupportedDataStores.JAEGER,
-  isDefault = false,
-  openSearch = {},
-  elasticApm = {},
-  signalFx = {},
-  jaeger = {},
-  tempo = {},
-  awsxray = {},
-  createdAt = '',
+  spec: {
+    id = '',
+    name = '',
+    type = SupportedDataStores.JAEGER,
+    default: isDefault = false,
+    createdAt = '',
+    openSearch = {},
+    elasticApm = {},
+    signalFx = {},
+    jaeger = {},
+    tempo = {},
+    awsxray = {},
+  } = {id: '', name: '', type: SupportedDataStores.JAEGER},
 }: TRawDataStore): DataStore => ({
   id,
```

Comment:
I like this, instead of having to use the new format we can keep using our internal models!